### PR TITLE
CTPPS reco update to use LHCInfoPer* records

### PR DESCRIPTION
**Pull request by Glitchmin**
Draft PR introducing usage of new records LHCInfoPer* in PPS proton reco and adding new reconstruction tests to run with 2023 data using generated LHCInfo and LHCInfoPer* payloads. It also adds new LHCInfoCombined class that is meant to be consumed instead of old LHCInfo and can be created using both LHCInfo and LHCInfoPer* objects.
Based on PR https://github.com/cms-sw/cmssw/pull/39686/commits/2dd927a920ae87be6ecd7d4f6e773cdcafc63837 , addresses comments left on this PR.

### CODE REVIEW

File: Calibration/PPSAlCaRecoProducer/python/ALCARECOPPSCalMaxTracks_cff.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+print("in file ALCARECOPPSCalMaxTracks_cff.py")
+
```
Comment (tvami) [inline Calibration/PPSAlCaRecoProducer/python/ALCARECOPPSCalMaxTracks_cff.py:2]: Please del this up at this point

File: Calibration/PPSAlCaRecoProducer/python/ALCARECOPPSCalMaxTracks_cff.py [hunk 2]
```diff
@@ -14,9 +16,12 @@
 from EventFilter.CTPPSRawToDigi.ctppsRawToDigi_cff import totemTimingRawToDigi as _totemTimingRawToDigi
 from EventFilter.CTPPSRawToDigi.ctppsRawToDigi_cff import ctppsPixelDigis as _ctppsPixelDigis
 
-ctppsDiamondRawToDigiAlCaRecoProducer  = _ctppsDiamondRawToDigi.clone(rawDataTag = 'hltPPSCalibrationRaw')
-totemTimingRawToDigiAlCaRecoProducer = _totemTimingRawToDigi.clone(rawDataTag = 'hltPPSCalibrationRaw')
-ctppsPixelDigisAlCaRecoProducer = _ctppsPixelDigis.clone(inputLabel = 'hltPPSCalibrationRaw')
+# ctppsDiamondRawToDigiAlCaRecoProducer  = _ctppsDiamondRawToDigi.clone(rawDataTag = 'hltPPSCalibrationRaw')
+# totemTimingRawToDigiAlCaRecoProducer = _totemTimingRawToDigi.clone(rawDataTag = 'hltPPSCalibrationRaw')
+# ctppsPixelDigisAlCaRecoProducer = _ctppsPixelDigis.clone(inputLabel = 'hltPPSCalibrationRaw')
```
Comment (tvami) [inline Calibration/PPSAlCaRecoProducer/python/ALCARECOPPSCalMaxTracks_cff.py:21]: please del

File: Calibration/PPSAlCaRecoProducer/test/2022_lhcinfo_test_AlCaRecoProducer.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+function die { echo $1: status $2; exit $2; }
+
+customise_commands="process.GlobalTag.toGet = cms.VPSet()\
+\nprocess.GlobalTag.toGet.append(cms.PSet(record = cms.string(\"CTPPSOpticsRcd\"),tag =  cms.string(\"PPSOpticalFunctions_test\"), connect = cms.string(\"sqlite_file:/afs/cern.ch/user/w/wcarvalh/public/CTPPS/optical_functions/validation/Aug22/PPSOpticalFunctions_singleIOV_2022_with_verticals.db\")))\
```
Comment (tvami) [inline Calibration/PPSAlCaRecoProducer/test/2022_lhcinfo_test_AlCaRecoProducer.sh:5]: these should be uploaded to the DB and have them read through the GT

File: Calibration/PPSAlCaRecoProducer/test/2023_lhcinfo_test_recoCTPPS_cfg.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+process.schedule = cms.Schedule(
+    process.path,
+    process.end_path
+)
+
+
+process.GlobalTag.toGet = cms.VPSet()
+process.GlobalTag.toGet.append(cms.PSet(record = cms.string("CTPPSOpticsRcd"),tag =  cms.string("PPSOpticalFunctions_2023_v1_validation"), connect = cms.string("frontier://FrontierProd/CMS_CONDITIONS")))
+process.GlobalTag.toGet.append(cms.PSet(record = cms.string("LHCInfoRcd"),tag =  cms.string("LHCInfo_PopCon_test"), connect = cms.string("sqlite_file:/afs/cern.ch/user/a/akulczyc/public/DBsRecoTestFill9019/lhcinfo_pop_unit_test_old.db")))
```
Comment (tvami) [inline Calibration/PPSAlCaRecoProducer/test/2023_lhcinfo_test_recoCTPPS_cfg.py:77]: same about the testing through the DB

File: CondTools/RunInfo/src/LHCInfoCombined.cc [hunk 5]
```diff
@@ -0,0 +1,51 @@
+#include "CondTools/RunInfo/interface/LHCInfoCombined.h"
+
+LHCInfoCombined::LHCInfoCombined(const LHCInfo& lhcInfo) { setFromLHCInfo(lhcInfo); }
```
Comment (tvami) [inline CondTools/RunInfo/src/LHCInfoCombined.cc:3]: the namespace should be lowercase, see 2.7 https://cms-sw.github.io/cms_coding_rules.html

File: RecoPPS/ProtonReconstruction/src/ProtonReconstructionAlgorithm.cc [hunk 6]
```diff
@@ -169,7 +169,7 @@ double ProtonReconstructionAlgorithm::newtonGoalFcn(double xi,
 //----------------------------------------------------------------------------------------------------
 
 reco::ForwardProton ProtonReconstructionAlgorithm::reconstructFromMultiRP(const CTPPSLocalTrackLiteRefVector &tracks,
-                                                                          const LHCInfo &lhcInfo,
+                                                                          const float energy,
```
Comment (grzanka) [inline RecoPPS/ProtonReconstruction/src/ProtonReconstructionAlgorithm.cc:172]: I'd encode the units in the variable name (if that is not against CMSSW coding rules), renaming `energy` to `energy_GeV` or `energy_TeV`

Comment (tvami) [reply RecoPPS/ProtonReconstruction/src/ProtonReconstructionAlgorithm.cc:172]: energy should be in GeV in CMSSW, if it's not then indeed better to include it in the name

Comment (grzanka) [reply RecoPPS/ProtonReconstruction/src/ProtonReconstructionAlgorithm.cc:172]: in PPS reconstruction code we use GeV, like in most places of CMSSW. I've made that suggestion to be more explicit, but we can stick to the GeV-convention as well.

File: CondTools/RunInfo/src/LHCInfoCombined.cc (hunk 5 above)
Comment (Glitchmin) [reply CondTools/RunInfo/src/LHCInfoCombined.cc:3]: I'm not defining/using any namespace here - LHCInfoCombined is a class

**Pull request closed without merging**